### PR TITLE
[DOCFIX] Add docs about locality with Spark + Yarn + Alluxio

### PR DESCRIPTION
**Pull request by aaudiber**

### CODE REVIEW

File: docs/en/Running-Spark-on-Alluxio.md [hunk 1]
```diff
@@ -135,3 +135,12 @@ in Spark WebUI below.
 
 ![locality]({{site.data.img.screenshot_datalocality_tasklocality}})
 
+### Running Spark on YARN
+
+When a Spark job is run on YARN, Spark launches its executors without taking data locality into account.
+Spark will then correctly take data locality into account when deciding how to distribute tasks to its
+executors. For example, if `host1` contains `blockA` and a job using `blockA` is launched on the yarn
```
Comment (gpang) [inline docs/en/Running-Spark-on-Alluxio.md]: yarn -> YARN

to be consistent with rest of doc?

File: docs/en/Running-Spark-on-Alluxio.md [hunk 2]
```diff
@@ -135,3 +135,12 @@ in Spark WebUI below.
...
+cluster with `--num-executors=1`, Spark might place the only executor on `host2` and have poor locality.
+However, if `--num-executors=2` and executors are started on `host1` and `host2`, Spark will be smart
+enough to prioritize placing the job on host1. To maximize the amount of locality your Spark jobs attain,
```
Comment (gpang) [inline docs/en/Running-Spark-on-Alluxio.md]: host1 -> `host1`

File: docs/en/Running-Spark-on-Alluxio.md [hunk 3]
```diff
@@ -135,3 +135,12 @@ in Spark WebUI below.
...
+you should use as many executors as possible, ideally 1-2 executors per node in the YARN cluster.
```
Comment (gpang) [inline docs/en/Running-Spark-on-Alluxio.md]: is `1-2` number special? or do we just want multiple per node?

Also, what is the assumption or recommended deployment of the Alluxio workers w.r.t. the YARN nodes?

**Comment by gpang** [comment]
@aaudiber Thanks for the improvements! I left a few questions.

**Comment by apc999** [comment]
should this go to branch 1.1 or master?

File: docs/en/Running-Spark-on-Alluxio.md (hunk 1 above)
Comment (aaudiber) [inline docs/en/Running-Spark-on-Alluxio.md]: done

File: docs/en/Running-Spark-on-Alluxio.md (hunk 2 above)
Comment (aaudiber) [inline docs/en/Running-Spark-on-Alluxio.md]: done

File: docs/en/Running-Spark-on-Alluxio.md (hunk 3 above)
Comment (aaudiber) [inline docs/en/Running-Spark-on-Alluxio.md]: `1-2` isn't special, we just want to cover as many nodes as possible. I've changed it to say at least one executor per node

Alluxio should be deployed on all computation nodes, I've added this

**Comment by aaudiber** [comment]
@apc999 good call, I've retargeted to https://github.com/Alluxio/alluxio/pull/3607

**Pull request closed without merging**